### PR TITLE
Fix all linter/clippy warnings

### PR DESCRIPTION
**Pull request by mnaamani**
Based on https://github.com/Joystream/joystream/pull/3970

Fix all clippy warnings, and re-enable aggressive linter checks before successful build.

I left a few unused imports in some benchmarking.rs files where code was commented out temporarily, to be re-enabled after carthage release.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202132419573087/1202537343542077) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by dobertRowneySr — APPROVED** [review]
- [x] `cargo clippy` gives no warnings
- [x] `cargo test` gives no failing test

There is just a conflict to be resolved

File: runtime-modules/working-group/src/tests/fixtures.rs [hunk 1]
```diff
@@ -661,17 +661,15 @@ impl TerminateWorkerRoleFixture {
...
         );
         assert_eq!(actual_result, expected_result);
 
-        if actual_result.is_ok() {
-            if actual_result.is_ok() {
-                assert!(!<crate::WorkerById<Test, DefaultInstance>>::contains_key(
-                    self.worker_id
-                ));
-            }
+        if actual_result.is_ok() && actual_result.is_ok() {
```
Comment (Lezek123) [inline runtime-modules/working-group/src/tests/fixtures.rs]:
```suggestion
        if actual_result.is_ok() {
```

**Review by Lezek123 — CHANGES_REQUESTED** [review]
There are still some warnings when I run:
```
cargo clippy --release --features runtime-benchmarks
```
(perhaps we should make this check part of the CI?)

File: runtime-modules/working-group/src/tests/fixtures.rs (hunk 1 above)
Comment (mnaamani) [reply runtime-modules/working-group/src/tests/fixtures.rs]: done in eb7e7556253695db2dd22ec1d437974758e26976

**Comment by mnaamani** [comment]
> There are still some warnings when I run:
> 
> ```
> cargo clippy --release --features runtime-benchmarks
> ```
> 
> (perhaps we should make this check part of the CI?)

Applied `clippy --fix` in dcbd616cd215be4ecae8204bc54c07f2500ce54e

Added the clippy check for benchmarks in 64d5bf433745e666f2fa171823b1446f281e8393

**Review by Lezek123 — APPROVED** [review]
LGTM

**Pull request merged**